Proper DeliveryAPI api error content is now forwarded instead of failing with null exception.

MiroKentico opened the pull request:
### Motivation

Fixes #326

All `DeliveryClient` methods that reads data from Delivery API now forward proper API error content instead of failing with null exception.

This PR code additionally introduces few unit tests refactorings/enhancements + non breaking constructors changes.

Sevitas reviewed (changes requested):
I have just few notes:
1. Have you considered creating another constructor for responses (eg. DeliveryItemListingResponse) with just an ApiResponse param? It could save us setting those null values in the client.
2. Why there is no _succes check_ for _GetItemAsync, GetTypeAsync, GetContentElementAsync, GetTaxonomyAsync_ methods?
3. I think it would be better to move _succes check_ before assignement to the content variable.

MiroKentico commented:
> I have just few notes:
> 
>     1. Have you considered creating another constructor for responses (eg. DeliveryItemListingResponse) with just an ApiResponse param? It could save us setting those null values in the client.
> 
>     2. Why there is no _succes check_ for _GetItemAsync, GetTypeAsync, GetContentElementAsync, GetTaxonomyAsync_ methods?
> 
>     3. I think it would be better to move _succes check_ before assignement to the content variable.

1. No, I haven't thought about these constructors for null values. I will think about it and reach you to discuss it.
2. The reason is, that implementation was already done in a way, that they already produced correct error response  ... and added unit tests prove it. I will think about putting that `IsSuccess` check, but if I will put it there, it would be more for some code standardization than pure necessity.
3. You are right, it would be better place (to avoid doing not necessary code in these error cases). I will move this check as close to actual response retrieval where the code place is sub-optimal (case that you wrote).

MiroKentico commented:
> > I have just few notes:
> > ```
> > 1. Have you considered creating another constructor for responses (eg. DeliveryItemListingResponse) with just an ApiResponse param? It could save us setting those null values in the client.
> > 
> > 2. Why there is no _succes check_ for _GetItemAsync, GetTypeAsync, GetContentElementAsync, GetTaxonomyAsync_ methods?
> > 
> > 3. I think it would be better to move _succes check_ before assignement to the content variable.
> > ```
> 
>     1. No, I haven't thought about these constructors for null values. I will think about it and reach you to discuss it.
> 
>     2. The reason is, that implementation was already done in a way, that they already produced correct error response  ... and added unit tests prove it. I will think about putting that `IsSuccess` check, but if I will put it there, it would be more for some code standardization than pure necessity.
> 
>     3. You are right, it would be better place (to avoid doing not necessary code in these error cases). I will move this check as close to actual response retrieval where the code place is sub-optimal (case that you wrote).



> I have just few notes:
> 
>     1. Have you considered creating another constructor for responses (eg. DeliveryItemListingResponse) with just an ApiResponse param? It could save us setting those null values in the client.
> 
>     2. Why there is no _succes check_ for _GetItemAsync, GetTypeAsync, GetContentElementAsync, GetTaxonomyAsync_ methods?
> 
>     3. I think it would be better to move _succes check_ before assignement to the content variable.



> > I have just few notes:
> > ```
> > 1. Have you considered creating another constructor for responses (eg. DeliveryItemListingResponse) with just an ApiResponse param? It could save us setting those null values in the client.
> > 
> > 2. Why there is no _succes check_ for _GetItemAsync, GetTypeAsync, GetContentElementAsync, GetTaxonomyAsync_ methods?
> > 
> > 3. I think it would be better to move _succes check_ before assignement to the content variable.
> > ```
> 
>     1. No, I haven't thought about these constructors for null values. I will think about it and reach you to discuss it.
> 
>     2. The reason is, that implementation was already done in a way, that they already produced correct error response  ... and added unit tests prove it. I will think about putting that `IsSuccess` check, but if I will put it there, it would be more for some code standardization than pure necessity.
> 
>     3. You are right, it would be better place (to avoid doing not necessary code in these error cases). I will move this check as close to actual response retrieval where the code place is sub-optimal (case that you wrote).

Points 1. and 3. are now implemented, for point 2. is opened discussion within SDK maintainer team.

Sevitas commented on Kentico.Kontent.Delivery/ContentItems/DeliveryItemResponse.cs:
```diff
@@ -24,5 +24,13 @@ internal DeliveryItemResponse(ApiResponse response, T item) : base(response)
         {
```
using System.Collections.Generic; - unused

MiroKentico replied:
Removed.

Sevitas reviewed (dismissed):
Just a nitpick, otherwise LGTM.
